friendshipper/src-tauri/src/engine/unreal/provider.rs

```diff
@@ -79,13 +82,30 @@ impl EngineProvider for UnrealEngineProvider {
 
     #[instrument(skip(self, status))]
     async fn send_status_update(&self, status: &RepoStatus) {
-        if self.is_editor_process_running() {
+        if self.is_editor_process_running()
+            && self
+                .can_handle_requests
+                .load(std::sync::atomic::Ordering::Relaxed)
+        {
             let client = reqwest::Client::new();
-            _ = client
+            let future = client
                 .post("http://localhost:8091/friendshipper-ue/status/update".to_string())
                 .json(status)
-                .send()
-                .await;
+                .send();
+
+            tokio::pin!(future);
```

Comment:
Extremely helpful forum post: https://users.rust-lang.org/t/how-to-check-if-a-future-is-immediately-ready/86401

Reply:
I'd even consider linking this in a comment directly